app/models/order_detail.rb

```diff
@@ -167,6 +167,10 @@ def self.all_movable
     .where("order_details.state NOT IN('canceled', 'reconciled')")
   end
 
+  def self.purchased
```

Comment:
This scope already exists on OrderDetail (albeit with a terrible name): `scope :ordered, joins(:order).merge(Order.purchased)`. I much prefer `purchased`. I would vote for changing the name of the scope and then setting up `ordered` as an alias.

It would probably be too big of a hassle at this point, but we could give a deprecation warning when using `ordered` over `purchased` and then go clean up the usages.